src/mapml/handlers/ContextMenu.js

```diff
@@ -723,13 +775,135 @@ export var ContextMenu = L.Handler.extend({
+        document.activeElement.shadowRoot.activeElement.innerHTML === this._layerMenu.children[0].innerHTML) { //"zoom to layer" on layermenu
+        this._layerMenu.children[1].focus();
+      } else if (!this._layerMenu.hasAttribute('hidden')) {
+        this._layerMenu.children[0].focus();
+      } else {
+        if (this.activeIndex > 0) {
+          let prevIndex = this.activeIndex - 1;
+          while (this._items[prevIndex].el.el.disabled) {
+            prevIndex--;
+            if (prevIndex < 0) {
+              prevIndex = this._items.length - 1;
+            }
+          }
+          this._setActiveItem(prevIndex);
+        } else {
+          this._setActiveItem(this._items.length - 1);
+        }
+      }
+    } else if (key === 40) { //down arrow
+      if (!this._coordMenu.hasAttribute('hidden') && 
+        (document.activeElement.shadowRoot == null ||
+        document.activeElement.shadowRoot.activeElement.innerHTML === this._coordMenu.children[2].innerHTML)) { //"map" on submenu
+        this._coordMenu.children[0].focus();
+      } else if (!this._coordMenu.hasAttribute('hidden') && 
+        document.activeElement.shadowRoot.activeElement.innerHTML === this._coordMenu.children[1].innerHTML) { //"extent" on submenu
+        this._coordMenu.children[2].focus();
+      } else if (!this._coordMenu.hasAttribute('hidden') && 
+        document.activeElement.shadowRoot.activeElement.innerHTML === this._coordMenu.children[0].innerHTML) { //"Location" on submenu
+        this._coordMenu.children[1].focus();
+      } else if (!this._layerMenu.hasAttribute('hidden') && 
+        document.activeElement.shadowRoot.activeElement.innerHTML === this._layerMenu.children[0].innerHTML){ //"zoom to layer" on layermenu
+        this._layerMenu.children[1].focus();
+      } else if (!this._layerMenu.hasAttribute('hidden')){
+        this._layerMenu.children[0].focus();  
+      } else {
+        if (this.activeIndex < this._items.length - 1) {
+          //edge case at index 0
+          if (!this.isRunned && this.activeIndex == 0 && !this._items[this.activeIndex].el.el.disabled){
+            this._setActiveItem(0);
+            this.isRunned = true;
+          } else { //edge case over
+            let nextIndex = this.activeIndex + 1;
+            while (this._items[nextIndex].el.el.disabled) {
+              nextIndex++;
+              if (nextIndex >= this._items.length) {
+                nextIndex = 0;
+              }
+            }
+            this._setActiveItem(nextIndex);
+          }
+        } else {
+          let nextIndex = 0;
+          while (this._items[nextIndex].el.el.disabled) {
+            nextIndex++;
+            if (nextIndex >= this._items.length) {
+              nextIndex = 0;
+            }
+          }
+          this._setActiveItem(nextIndex);
+        }
+      }
+    } else if (key === 39) { //right arrow
+      if (document.activeElement.shadowRoot == null) { //do nothing when null
```

Comment:
This looks funny.  I would combine the negative logic with the logic in your else and eliminate the do-nothing.  The comment should also say **why** shadowRoot is null.